bson/src/main/org/bson/codecs/pojo/PojoCodec.java

```diff
@@ -194,6 +194,9 @@ private void decodeFields(final BsonReader reader, final DecoderContext decoderC
             codec = new CollectionCodec(head, getCodecFromTypeData(typeData.getTypeParameters().get(0)));
         } else if (Map.class.isAssignableFrom(head)) {
             codec = new MapCodec(head, getCodecFromTypeData(typeData.getTypeParameters().get(1)));
+        } else if (Enum.class.isAssignableFrom(head)) {
```

Comment:
Added a registry check to automatically do this. Making users set the codec explicitly for a field would probably be too much.